add a new `toolkit install` command with `--features` support

amtoine opened the pull request:
# Description
i was installing Nushell and, as we have the `dataframe` feature and very soon at least the `extra` feature with more and more commands, i thought it could be cool to have a little `toolkit install` command :yum: 

# User-Facing Changes
exposes the following command to developers
```
install Nushell and features you want

Usage:
  > install ...(features) 

Flags:
  -h, --help - Display the help message for this command

Parameters:
  ...features <string>: a space-separated list of feature to install with Nushell
```

# Tests + Formatting
- :green_circle: `toolkit fmt`
- :green_circle: `toolkit clippy`
- :black_circle: `toolkit test`
- :black_circle: `toolkit test stdlib`

# After Submitting
```
$nothing
```

fdncred commented:
I think this is a good idea too. In fact, I was thinking that maybe all the nushell scripts in the scripts folder should be apart of toolkit. Thoughts?

amtoine commented:
> I think this is a good idea too.
>
:partying_face: 

> In fact, I was thinking that maybe all the nushell scripts in the scripts folder should be apart of toolkit. Thoughts?
>
... :eyes: 
guess what... :eyes: 

i've thought about that exact same thing when we talked about the root refactoring :grimacing: 
and i agree we might want to break users once more, for the greater good :muscle: :yum: 


my questions would just be
- in this PR
- what about the non-`nu` scripts

fdncred commented:
i'm fine with putting it in this PR or another. it's up to you. I'd also leave the nu scripts in place for backwards compatibility for a release or two.

the other scripts should stay.

amtoine commented:
@fdncred 
i've simply
- added an `std log warning` in the `.nu` scripts to mention the future deprecation of these scripts
- moved and adapted the scripts to the `toolkit` as
  - `toolkit build [--all]`
  - `toolkit cov`
  - `toolkit plugin register`

:yum:

fdncred commented on toolkit.nu:
```diff
@@ -274,4 +316,91 @@ export def install [
+}
+
+# register all installed plugins
+export def "plugin register" [] {
```
How about naming it this?
```suggestion
export def "register plugins" [] {
```

amtoine replied:
no strong preference on this :+1: 
that looks more like the original script so let's go with this :wink:

amtoine replied:
done in e00323e56372a81bf494f05eb51b03a005d5451f

fdncred commented:
I think it looks good except for the one custom command rename I suggested. Thanks!

amtoine commented:
thanks @fdncred for the review, let's wait for the CI and land this then :blush:

fdncred commented:
I was wanting to rebuild and install the plugins. I found `toolkit build --all` but I found no way to install the plugins other than using the install scripts install-all.sh or install-all.ps1. It would be nice if these were added to toolkit as well as uninstall-all.sh functionality.

fdncred commented:
> i think there has never been such an "install all" script written in nu, right? 

No, there hasn't been. It's always been sh and ps1.

amtoine commented:
> No, there hasn't been. It's always been sh and ps1.
>
i see :+1: 

i think the way right now is to
- `toolkit build --all`
- `toolkit install ...(features)`
- `toolkit register plugins`

can you tell me if that works? :blush:

fdncred commented:
If I'm reading your code correctly, this is what is happening.
- `toolkit build --all` will build nu and the plugins in debug mode but not release mode
- `toolkit install --features=$features` will build and install nushell in release mode but not the plugins
- `toolkit register plugins` will register plugins regardless of debug/release mode if they're in your path

What's missing is an install command that builds the plugins in release mode and installs them like `cargo install --path .` in each plugin dir.

amtoine commented:
i do not have the time tonight so i've filed an issue for this :wink:
